client/src/main/java/org/asynchttpclient/netty/channel/NonBlockingSemaphoreInfinite.java

```diff
@@ -0,0 +1,21 @@
+package org.asynchttpclient.netty.channel;
+
+/**
+ * Non-blocking semaphore-like object with infinite permits.
+ *
+ * So try-acquire always succeeds.
+ *
+ * @author Stepan Koltsov
+ */
+enum NonBlockingSemaphoreInfinite implements NonBlockingSemaphoreLike {
```

Comment:
No-op does not make it clear that `tryAcquire` returns true. `Infinite` make it look like a semaphore with infinite number of permits, so `tryAcquire` always returns true.

I put `Infinite` part at the end of the class name so classes were together in the file list in the IDE.

So I think names are good, but I can change both name and placement if you still think it is better.
<img width="353" alt="2017-03-28_2023" src="https://cloud.githubusercontent.com/assets/28969/24418394/93903754-13f4-11e7-8b8d-80b9dbc93f2e.png">